### PR TITLE
adjust writers fmt AND fix TestCreateProcessContainer AND add length list check

### PR DESCRIPTION
**Pull request by fahedouch**
this PR:

* adjust fmt writers
* add length list check
* fix https://github.com/containerd/nerdctl/issues/2458
----------------------------
this PR do not handle :  `Why container  is created w/o a name assigned ?` ([seen in TestCreateProcessContainer)](https://github.com/containerd/nerdctl/actions/runs/5952944195/job/16146057148?pr=2447)

### CODE REVIEW

**Review by yankay — APPROVED** [review]

**Comment by AkihiroSuda** [comment]
In this case you are squashing commits too much.
1 commit = 1 topic.

File: cmd/nerdctl/container_create_windows_test.go [hunk 1]
```diff
@@ -36,7 +35,6 @@ func TestCreateProcessContainer(t *testing.T) {
 }
 
 func TestCreateHyperVContainer(t *testing.T) {
-	//t.Parallel()
```
Comment (AkihiroSuda) [inline cmd/nerdctl/container_create_windows_test.go:39]: Should be a separate commit

File: cmd/nerdctl/container_list.go [hunk 2]
```diff
@@ -198,10 +198,14 @@ func formatAndPrintContainerInfo(containers []container.ListItem, options Format
...
 		} else if options.Quiet {
-			if _, err := fmt.Fprintf(w, "%s\n", c.ID); err != nil {
+			if _, err := fmt.Fprintln(w, c.ID); err != nil {
 				return err
 			}
 		} else {
+			var name string
+			if len(c.Names) > 0 {
+				name = c.Names[0]
+			}
```
Comment (AkihiroSuda) [inline cmd/nerdctl/container_list.go:208]: Should be a separate commit

File: cmd/nerdctl/container_list.go [hunk 3]
```diff
@@ -198,10 +198,14 @@ func formatAndPrintContainerInfo(containers []container.ListItem, options Format
 				return err
 			}
 		} else if options.Quiet {
-			if _, err := fmt.Fprintf(w, "%s\n", c.ID); err != nil {
+			if _, err := fmt.Fprintln(w, c.ID); err != nil {
```
Comment (AkihiroSuda) [inline cmd/nerdctl/container_list.go:201]: This should be in the third commit, not in the first one

Comment (fahedouch) [reply cmd/nerdctl/container_list.go:201]: @AkihiroSuda Sorry for that, Fixed :)

**Review by AkihiroSuda — APPROVED** [review]
Thanks
